core/kazoo_number_manager/src/knm_number_states.erl

```diff
@@ -234,11 +234,17 @@ in_service_from_reserved_authorize(Number) ->
 in_service_from_in_service_authorize(Number) ->
     PhoneNumber = knm_number:phone_number(Number),
     AssignTo = knm_phone_number:assign_to(PhoneNumber),
+    AssignedTo = knm_phone_number:assigned_to(PhoneNumber),
     AuthBy = knm_phone_number:auth_by(PhoneNumber),
     Sudo = ?KNM_DEFAULT_AUTH_BY =:= AuthBy,
     case Sudo
-        orelse ?ACCT_HIERARCHY(AssignTo, AuthBy, 'true')
```

Comment:
Could you try and see if just using knm_phone_number:is_authorized/1 solves your problem here?
There's a bunch of different ways auth is verified in this app and I'm trying to reduce them all in just one place.